Fullscript/eslint-plugin
# Add Typescript

## Description

This is the PR to add TypeScript into the codebase. 

- Added `typescript` pkg and some `@types` pkgs
- Most files are simply converted from `.js` to `.ts` . I don't touch any logic

## Test

- Run `yarn type-check` in your local and make sure there are no type errors
- Run `yarn test` in your local and make sure all tests pass
- Run `yarn build` in your local and make sure no diffs happen
- Check [changed files](https://github.com/Fullscript/eslint-plugin/pull/14/files) and make sure no `dist` changes expect for some spec files

## TODO

I'm not sure where we should put `yarn type-check` to check type errors. Should we put it in the `build` script like this?

```
// package.json

...
"build": "yarn type-check && swc ./src -d dist"
...
```

Or should we put it in [this husky shell file](https://github.com/Fullscript/eslint-plugin/blob/main/.husky/prettier_build_check.sh)?
